src/app/workflow/view/view.service.ts

```diff
@@ -176,11 +212,16 @@ export class ViewService {
    */
   requestDOIForWorkflowVersion(workflow: Workflow, version: WorkflowVersion): void {
     // Set the dialog to open a snapshot if its not frozen
-    // TODO check to see if they have a zenodo token
-    if (!version.frozen) {
-      this.showSnapshotBeforeDOIDialog(workflow, version);
-    } else {
-      this.showRequestDOIDialog(workflow, version);
-    }
+    this.tokenQuery.hasZenodoToken$.subscribe((hasZenodoToken: boolean) => {
```

Comment:
You need to either unsubscribe or probably better yet do something like `pipe(first())` or `pipe(take(1)`.

I'm also a little unclear if each new subscription gets the current value, although it probably does... Did you try for more than one version on the same page?